File: StudentGradeCalculator.py

def calculate_grade(avg):
    if avg >= 90:
        return 'A'
    elif avg >= 80:
        return 'B'
    elif avg >= 70:
        return 'C'
    elif avg >= 60:
        return 'D'
    else:
        return 'F'


def main():
    student_name = input("Enter student name: ")
    num_subjects = int(input("Enter number of subjects: "))

    marks = []
    total = 0

    for i in range(num_subjects):
        mark = float(input(f"Enter marks for subject {i+1}: "))
        marks.append(mark)
        total += mark

    average = total / num_subjects
    grade = calculate_grade(average)

    print("\n--- Result ---")
    print(f"Student Name: {student_name}")
    print(f"Total Marks: {total}")
    print(f"Average Marks: {average:.2f}")
    print(f"Grade: {grade}")

if __name__ == "__main__":
    main()

File: StudentGradeCalculator.cpp
#include <iostream>
#include <string>
using namespace std;

// Function to calculate grade based on average
char calculateGrade(float avg) {
    if (avg >= 90) return 'A';
    else if (avg >= 80) return 'B';
    else if (avg >= 70) return 'C';
    else if (avg >= 60) return 'D';
    else return 'F';
}

int main() {
    string studentName;
    int numSubjects;

    cout << "Enter student name: ";
    getline(cin, studentName);

    cout << "Enter number of subjects: ";
    cin >> numSubjects;

    float marks[numSubjects];
    float total = 0;

    for (int i = 0; i < numSubjects; i++) {
        cout << "Enter marks for subject " << i + 1 << ": ";
        cin >> marks[i];
        total += marks[i];
    }

    float average = total / numSubjects;
    char grade = calculateGrade(average);

    cout << "\n--- Result ---\n";
    cout << "Student Name: " << studentName << endl;
    cout << "Total Marks: " << total << endl;
    cout << "Average Marks: " << average << endl;
    cout << "Grade: " << grade << endl;

    return 0;
}
